### ISSUE TITLE
Please add basic prediction example to transfer learning guide

### ISSUE THREAD

**Issue by ApproximateIdentity** [keras-team/keras-io#463]
Unless I'm missing something basic, I don't see any example of actually using the transfer learning guide to make a prediction

https://github.com/keras-team/keras-io/blob/master/guides/transfer_learning.py
https://keras.io/guides/transfer_learning/

For example, if I run all the code in the "end-to-end" example at the end, my naive attempt at making some form of prediction is the following:

```
for obj, _ in train_ds:
    print(model.predict(obj))
    break

[[ 5.7956142]
 [ 7.138669 ]
 [ 6.004242 ]
 [-7.5267677]
 [ 4.4232373]
 [ 4.33535  ]
 [-6.3878946]
 [-5.303356 ]
 [ 4.605398 ]
 [ 8.426602 ]
 [10.56328  ]
 [-7.899274 ]
 [-9.870649 ]
 [ 3.5054545]
 [-9.1837   ]
 [-8.082974 ]
 [-9.05807  ]
 [ 7.5205717]
 [-6.4828777]
 [ 8.651134 ]
 [ 7.1191254]
 [-5.9992075]
 [ 5.146449 ]
 [ 4.319687 ]
 [-8.059815 ]
 [-8.574115 ]
 [ 9.458901 ]
 [ 6.465415 ]
 [ 6.0897965]
 [10.061241 ]
 [10.10572  ]
 [ 3.1187348]]
```

I'm quite confused as to what these values represent. I believe the `base_model` had more classes than just cat or dog and so maybe it's giving me all the classes. That's just speculation anyway. I think there should be a final example that shows an actual prediction resulting in a value "cat" or "dog" to complete this example. To me this seems like maybe the single most important piece of everything.

Anyway I'll probably sort this out myself, but if someone reading this can explain how to make predictions here I'd be grateful (and I could even take it upon myself to expand the `transfer_learning.py` guide myself).

**Comment by sayannath** [comment]
Hi @ApproximateIdentity, you can refer to [this](https://keras.io/examples/vision/image_classification_efficientnet_fine_tuning/)

Hope you find this useful, if not, I can help you with more resources.

Thank You!

**Comment by SuryanarayanaY** [comment]
Hi @ApproximateIdentity ,

Thanks for pointing that. Actually the training example is for cat and dog classification hence there should be `sigmoid` activation at the last layer which is missing in the code. Also `from_logits=False` makes more sense for the loss function as it is binary classification. 

I tried the model with above changes and now `model.predict(train_ds.take(1))` now outputs 32 values which ranges in between 0 to 1 (>0.5 can be transformed as 1 i.e Dog and <=0.5 can be taken as 0,i.e Cat ) . We are getting 32 outputs since our input is a dataset of `batch_size=32` (default size). Please refer attached [gist](https://colab.research.google.com/gist/SuryanarayanaY/8a57cabc653aa74a127ad44a6cffd4c9/kerasio_463-transfer_learning.ipynb#scrollTo=iP-mv3O_QILD).

Hope this will clarify your doubt.

I will also create a PR for above changes.

Thanks.

**Comment by MarkDaoust** [comment]
Hi, @SuryanarayanaY thanks for tanking the time to make a PR.

The interpretation of the outputs is negative values are class zero, positive values are class one. you can always run it through a sigmoid yourself.

There are good arguments for both ways of doing it. 

There's nothing wrong with using `activation=None + from_logits=True`. This approach avoids some numerical problems that can break `activation="sigmoid" + from_logits=False`. Keras usually handles that transparently.  If you're using eager execution keras can't apply that fix. In some cases it will have almost no effect (since keras does use an `eps` on the loss calculation so you won't get NaNs), but in some cases (if you have a large number of classes) the effect could be important. 

Anyway, there's no clear standard here, in this repository:

- [from_logits=True](https://github.com/search?q=repo%3Akeras-team%2Fkeras-io++from_logits%3DTrue&type=code) has ~170 uses.
- [activation='sigmoid'](https://github.com/search?q=repo%3Akeras-team%2Fkeras-io++activation%3D%22sigmoid%22&type=code) or [activation='softmax'](https://github.com/search?q=repo%3Akeras-team%2Fkeras-io++activation%3D%22softmax%22&type=code) has ~210.

So I'm -1 on changing just one for personal preferences.
